src/ring_theory/hahn_series.lean

```diff
@@ -822,6 +822,23 @@ end
 
 end valuation
 
+lemma is_pwo_Union_support_powers [linear_ordered_add_comm_group Γ] [integral_domain R]
+  {x : hahn_series Γ R} (hx : 0 < add_val Γ R x) :
+  (⋃ n : ℕ, {g | (x ^ n).coeff g ≠ 0 }).is_pwo :=
```

Comment:
wasn't there a dedicated definition `(x ^ n).support` to talk about the support?

Reply:
Yes, there is.